### DISCUSSION TITLE
v14: Async Healthchecks

### DISCUSSION THREAD

**Discussion by mattbrailsford** [umbraco/Umbraco-CMS#17397] [Features and ideas]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

v14

### Bug summary

The current health checks implementation only supports async when fetching the status, however the execute action behaviour of health checks only supports running synchronously.

### Specifics

In Umbraco Commerce v15 our who API has been made async and so we need to be able to execute our async services in response to health check interactions.

### Steps to reproduce

n/a

### Expected result / actual result

I expect to be able to execute async code in response to a health check interaction

**Comment by AndyButland** [comment]
Fixed with https://github.com/umbraco/Umbraco-CMS/pull/17090 and due in 15.3.
